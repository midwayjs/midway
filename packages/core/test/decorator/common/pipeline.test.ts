import {
  Pipeline,
  Provide,
  INJECT_CUSTOM_PROPERTY,
  MetadataManager
} from '../../../src';

@Provide()
class Test {
  @Pipeline()
  dd: any;
}

describe('/test/annotation/pipeline.test.ts', () => {
  it('pipeline decorator should be ok', () => {
    expect(MetadataManager.getPropertiesWithMetadata(INJECT_CUSTOM_PROPERTY, Test)).toMatchSnapshot();
  });
});
