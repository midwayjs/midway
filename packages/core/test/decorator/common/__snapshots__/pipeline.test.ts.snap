// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/test/annotation/pipeline.test.ts pipeline decorator should be ok 1`] = `
{
  "dd": [
    {
      "key": "__pipeline_identifier__",
      "metadata": {
        "valves": undefined,
      },
      "options": {
        "impl": true,
      },
      "propertyName": "dd",
    },
  ],
}
`;
